src/vellum/workflows/nodes/core/retry_node/node.py

```diff
@@ -53,60 +56,90 @@ def run(self) -> BaseNode.Outputs:
                     node_output_mocks=self._context._get_all_node_output_mocks(),
                 )
 
-            node_outputs: Optional[BaseNode.Outputs] = None
-            exception: Optional[NodeException] = None
-            for event in subworkflow_stream:
-                self._context._emit_subworkflow_event(event)
-
-                if not is_terminal_workflow_execution_event(event):
-                    continue
-
-                if event.workflow_definition != self.subworkflow:
-                    continue
-
-                if event.name == "workflow.execution.fulfilled":
-                    node_outputs = self.Outputs()
-
-                    for output_descriptor, output_value in event.outputs:
-                        setattr(node_outputs, output_descriptor.name, output_value)
-                elif event.name == "workflow.execution.paused":
-                    exception = NodeException(
-                        code=WorkflowErrorCode.INVALID_OUTPUTS,
-                        message=f"Subworkflow unexpectedly paused on attempt {attempt_number}",
-                    )
-                elif self.retry_on_error_code and self.retry_on_error_code != event.error.code:
-                    exception = NodeException(
-                        code=WorkflowErrorCode.INVALID_OUTPUTS,
-                        message=f"""Unexpected rejection on attempt {attempt_number}: {event.error.code.value}.
-Message: {event.error.message}""",
-                    )
-                elif self.retry_on_condition and not resolve_value(self.retry_on_condition, self.state):
-                    exception = NodeException(
-                        code=WorkflowErrorCode.INVALID_OUTPUTS,
-                        message=f"""Rejection failed on attempt {attempt_number}: {event.error.code.value}.
-Message: {event.error.message}""",
-                    )
+            if self.timeout is None:
+                o = self._run_subworkflow(subworkflow_stream, attempt_number)
+                if isinstance(o, NodeException):
+                    last_exception = o
                 else:
+                    return o
+            else:
+                try:
+                    with multiprocessing.pool.ThreadPool(processes=1) as pool:
+                        async_result = pool.apply_async(
+                            self._run_subworkflow, args=(subworkflow_stream, attempt_number)
```

Comment:
[oos] The subworkflow will already have a thread that initiates, I wonder if we could pass the timeout directly to the workflow